src/context/LocationsContext.js

```diff
@@ -26,8 +32,28 @@ const LocationsContextProvider = (props) => {
       .catch(console.log);
   }, []);
 
+  /*useEffect for when the recenter changes which happens in the map component, 
+  the api is call the api & the new locations are set */
+  useEffect(() => {
+    const recenterUrl =
+      recenter &&
+      `https://segdeha.com/api/nearby.php?lat=${recenter.lat}&lng=${recenter.lng}`;
+
+    fetch(recenterUrl)
+      .then((response) => {
+        return response.json();
+      })
+      .then((response) => {
+        let pages = response.query ? response.query.pages : [];
+        setLocations(pages);
+      })
+      .catch(console.log);
+  }, [recenter]);
+
   return (
-    <LocationsContext.Provider value={{ locations, coordinates, zoomLevel }}>
+    <LocationsContext.Provider
+      value={{ locations, coordinates, zoomLevel, recenter, setRecenter }}
```

Comment:
`zoomLevel` is only needed within `Map.js` ... Would it be possible to keep it encapsulated there?